librz/main/rz-find.c

```diff
@@ -190,6 +195,56 @@ static int rzfind_open_file(RzfindOptions *ro, const char *file, const ut8 *data
 		return 0;
 	}
 
+	if (ro->import || ro->symbol) {
+		RzBinFile *bf;
+		const RzList *symbols;
+		RzListIter *iter, *it;
+		RzBinSymbol *symbol;
+		RzBin *bin = rz_bin_new();
+		RzIO *rio = rz_io_new();
+		RzBinOptions opt = { 0 };
+
+		if (!bin || !rio) {
+			result = 1;
+			goto sym_end;
+		}
+
+		rz_io_bind(rio, &bin->iob);
+		rz_bin_options_init(&opt, 0, 0, 0, false, false);
+
+		bf = rz_bin_open(bin, file, &opt);
+		if (!bf) {
+			result = 1;
+			goto sym_end;
+		}
+
+		symbols = rz_bin_object_get_symbols(bf->o);
+		rz_list_foreach (ro->keywords, iter, kw) {
+			rz_list_foreach (symbols, it, symbol) {
+				if (!symbol->name) {
+					continue;
+				}
+
+				if (ro->import && symbol->is_imported && !strcmp(symbol->name, kw)) {
+					printf("ordinal: %d %s\n", symbol->ordinal, kw);
+					break;
+				}
+
+				if (ro->symbol && !strcmp(symbol->type, "FUNC") && !strcmp(symbol->name, kw)) {
+					printf("paddr: 0x%08" PFMT64x " vaddr: 0x%08" PFMT64x " %s\n", symbol->paddr, symbol->vaddr, symbol->name);
+					break;
+				}
```

Comment:
There is `rz_bin_object_get_imports` for the imports. I would avoid doing this kind of filtering and just rely on the list provided by either of these functions.

Reply:
Remove the filter part. Thanks for your reminder.